src/UnitOfWork.php

```diff
@@ -275,7 +300,7 @@ private function manageEntityReference($oid)
         $this->entityStateReferences[$id] = clone $entity;
     }
 
-    private function detectEntityChanges()
+    public function detectEntityChanges()
```

Comment:
Makes sense, thanks for clarifying. I'm happy to merge but be aware that during refactoring there can be some breaking changes, I'll keep you notified anyway if changes can affect you.